libsel4/sel4_arch_include/aarch32/sel4/sel4_arch/constants.h

```diff
@@ -130,9 +130,6 @@ enum {
 #ifdef CONFIG_KERNEL_MCS
 enum {
     seL4_Timeout_Data,
-    /* consumed is 64 bits */
```

Comment:
"
The new model of time-out faults assumes that a time-out handler is
associated with thread (or at least the thread's time-out endpoint) and
does not assume that the time-out handler has any control over an SC.
"

Timeout faults were primarily designed for either user level scheduling or to handle client SC's running out of time in a shared server. In the latter case, in order to implement different escape strategies for the server, I needed to be able to  identify which SC had faulted. In the former case, I needed the consumed value to do efficient user level scheduling. 

Why is this being removed? How will you handle the rollback scenarios identified in my thesis? How much longer will user level scheduling take now that you have to make an extra invocation to get the consumed amount?

Reply:
What was your reason for re-setting the consumed time in the SC when it is retrieved by syscall or fault?

The case that I am trying to preserve is having a passive server reply to a client such that it faults again and having the timeout fault handler for the client not observe a 0 consumed time in the fault.

It looks like you are trying to handle the case where the timer resolution is more coarse than 1us and so conversion to us may lead to overflow. Given that system time will eventually overflow both in ticks and in us, we could simply only guarantee that the system is guaranteed to behave correctly up to the lower of either bound.

Would it be reasonable to not reset the consumed time when providing it to user level and simply accept any overflow that might occur in the process?

Reply:
> What was your reason for re-setting the consumed time in the SC when it is retrieved by syscall or fault?

If you are using this info you are most likely trying to monitor the execution of the thread from user level. I couldn't see any other more efficient way to figure out what you had already read and get accurate accumluations over time. Look at the implementation of a user-level CFS scheduler. 

> The case that I am trying to preserve is having a passive server reply to a client such that it faults again and having the timeout fault handler for the client not observe a 0 consumed time in the fault.

I assumed the fault handlers were per server and cared about the time spent in the server. 

> It looks like you are trying to handle the case where the timer resolution is more coarse than 1us and so conversion to us may lead to overflow.

No. Although it does help. 

> Would it be reasonable to not reset the consumed time when providing it to user level and simply accept any overflow that might occur in the process?

No.